src/NameGenerator.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\StorageDriver\BigQuery;
+
+class NameGenerator
+{
+    protected string $stackPrefix;
+
+    public function __construct(
+        string $stackPrefix
+    ) {
+        $this->stackPrefix = $stackPrefix;
+    }
+
+    public function createProjectId(string $projectId): string
+    {
+        return sprintf('%s-%s', $this->stackPrefix, $projectId);
+    }
```

Comment:
napsal bych do readme, že limit je 30zn já mám `zajca-dev-project-10-17-12-14-06` :D a je to 32zn dostanu pak:
```
Google\ApiCore\ApiException: {
    "message": "Request contains an invalid argument.",
    "code": 3,
    "status": "INVALID_ARGUMENT",
    "details": [
        {
            "@type": "type.googleapis.com\/google.rpc.BadRequest",
            "fieldViolations": [
                {
                    "field": "project.project_id",
                    "description": "project_id must be at most 30 characters long"
                },
                {
                    "field": "project.display_name",
                    "description": "project display name must be at most 30 characters"
                }
            ]
        }
    ]
}

```

Reply:
jak sme sa bavili, tak by to mohlo byt by default local, tím že je to scopované tak to nemusíme řešit a nedává moc smysl aby si to každý nastavoval jinak.

Reply:
pridal som ze ak nemas nastaveny ten prefix v env tak je to automaticky `local` plus som este pridal info ohladne toho ake moze byt to projectId https://github.com/keboola/php-storage-driver-bigquery/pull/7/commits/1cc64e57f87ffc228a52b04a1ea2a58f1a787be1

Reply:
aha to je unikátní cross celý cloud :thinking: tam budem muset přidat nějaký hash nebo něco. Protože stack prefix je teď všude `keboola` bych řekl. Takže sa teoreticky možů konfliktit vzájemně úplně oddělené věci. Třeba dva klienti na private stacku s BQ co chců založit projekt id 1.

Reply:
fakt ? som myslel ze kazdy stack ma svoj spesl prefix, hmm

Reply:
No právě bych řekl že teď je všude KEBOOLA @Halama je to tak nebo ne?

Reply:
Jo nové produkční stacky mají typicky prefix `keboola_` https://github.com/keboola/kbc-stacks/search?q=CONNECTION_CLIENT_DB_PREFIX 
<img width="871" alt="image" src="https://user-images.githubusercontent.com/903531/196414450-558df6f7-1dce-4179-aebd-4cf753b4ab36.png">

Reply:
> aha to je unikátní cross celý cloud 🤔 tam budem muset přidat nějaký hash nebo něco. Protože stack prefix je teď všude `keboola` bych řekl. Takže sa teoreticky možů konfliktit vzájemně úplně oddělené věci. Třeba dva klienti na private stacku s BQ co chců založit projekt id 1.

je to teda na urovni org, takze pokial su obaja klienti v inej org tak je to v pohode, tzn ze by to zatial nepodporovalo multitenant. Len premyslam ze moc neviem jak tam nacpat nejaky unikatny prefix bez toho aby som este nieco pridaval do connection

Reply:
Tak to teď asi stačí, když to enom pro org.
Tobě z connection dojde stack_prefix a project_id, ale přidat si možeš co chceš, driver řídí to jak bude pojmenovaný resource takže když to bude `<stack_prefix>_<project_id>_<hash>` tak je to OK, ty to enom pak vrátíš, žes něco takového vytvořil a connection to uloží do metadat.